app/translations/en.json

```
@@ -542,7 +542,8 @@
                 "newKey": "New key",
                 "newValue": "New value",
                 "valuePlaceholder": "Select or type in a value",
-                "validationError": "Only letters, digits and the characters \"-\", \".\" and \"_\" are allowed and it must begin with a letter.",
+                "keyValidationError": "Only letters, digits and the characters \"-\", \".\" and \"_\" are allowed.",
+                "valueValidationError": "All characters except \" are allowed.",
```

Minor:

The error tells the user what's allowed, not what they did wrong. I would rather be told what is not allowed, than what's allowed:

```suggestion
                "valueValidationError": "The \" character is not allowed.",
```

---

Make sense, I also added info about control characters, which are not allowed, too.

---

Good idea :+1: Although it would probably be pretty rare that someone will use control characters :smile: Maybe that would just confuse the user? I'm not sure, but whatever, let's leave it :smile:

---

Yeah, I was debating with myself about adding/not adding it having the same concern - "Maybe that would just confuse the user?"

But then I received such question from Sandy on Slack yesterday:
> Out of interest (not required information or anything, I'm just interested), if we're supporting all utf-8 for labels, what do you do if someone puts say a right-to-left special character (that switches text direction after that point) in a label?

so I think it make sense to be more informative 🙂

---

Oh, I see :smile: Makes sense now

One way we could get the best of both worlds (be informative when there are invalid control characters and do not confuse the user otherwise) would be to use different error messages for those 2 distinct cases :smile: But that is a minor issue